### PR TITLE
Added GranularAccessScope request to AccessScopeService

### PR DESCRIPTION
**Pull request by lucasabond**
Following up on issue 840 https://github.com/nozzlegear/ShopifySharp/issues/840.

### CODE REVIEW

File: ShopifySharp/Services/AccessScope/AccessScopeService.cs [hunk 1]
```diff
@@ -20,7 +22,31 @@ public class AccessScopeService : ShopifyService, IAccessScopeService
...
-            await ExecuteGetAsync<IEnumerable<AccessScope>>("oauth/access_scopes.json", "access_scopes", cancellationToken: cancellationToken);
+        public virtual async Task<IEnumerable<AccessScope>> ListAsync(CancellationToken cancellationToken = default)
+        {
+            return await ExecuteGetAsync<IEnumerable<AccessScope>>("oauth/access_scopes.json", "access_scopes", cancellationToken: cancellationToken);
+        }
+
+        /// <summary> 
+        ///  Requests a subset of granular access scopes for an individual shop installation. 
+        /// </summary> 
+        public virtual async Task<IEnumerable<AccessScope>> RequestGranularAccessScopesAsync(IEnumerable<AuthorizationScope> requestedScopes, CancellationToken cancellationToken = default)
```
Comment (nozzlegear) [inline ShopifySharp/Services/AccessScope/AccessScopeService.cs:33]: Looks good! Just need to add the two new methods to the service interface.

File: ShopifySharp/Services/AccessScope/AccessScopeService.cs [hunk 2]
```diff
@@ -20,7 +22,31 @@ public class AccessScopeService : ShopifyService, IAccessScopeService
...
+        /// </summary> 
+        public virtual async Task<IEnumerable<AccessScope>> RequestGranularAccessScopesAsync(IEnumerable<AuthorizationScope> requestedScopes, CancellationToken cancellationToken = default) 
+        {
+            return await RequestGranularAccessScopesAsync(requestedScopes.Select(s => s.ToSerializedString()), cancellationToken); 
+        }
+
+        /// <summary> 
+        ///  Requests a subset of granular access scopes for an individual shop installation. 
+        /// </summary> 
+        public virtual async Task<IEnumerable<AccessScope>> RequestGranularAccessScopesAsync(IEnumerable<string> requestedScopes, CancellationToken cancellationToken = default)
```
Comment (nozzlegear) [inline ShopifySharp/Services/AccessScope/AccessScopeService.cs:41]: Same here, this just needs to be added to the IAccessScopeService interface and we'll be good to go!

**Review by nozzlegear — CHANGES_REQUESTED** [review]

**Comment by nozzlegear** [comment]
Awesome, thanks for the pull request! I just need one small change, can you please add the two new methods to the IAccessScopeService interface? Once that's done I'll get this merged and published.

**Comment by lucasabond** [comment]
Just pushed the additions to the interface. Glad to be able to help. I think this is all I have for now, but will push any other additions I build out. Cheers.

**Review by nozzlegear — APPROVED** [review]

**Pull request merged**